src/bridge/session.rs

```diff
@@ -0,0 +1,112 @@
+//! This module contains adaptations of the functions found in
+//! https://github.com/KillTheMule/nvim-rs/blob/master/src/create/tokio.rs
+
+use core::result::Result as CoreResult;
```

Comment:
One last thing that surprises me a bit is that you're using `core::result::Result` here. `std` re-exports many types from `core`, including `Result`.